### DISCUSSION TITLE
如何给中英文摘要和目录添加页码

### DISCUSSION THREAD

**Discussion by MZGuo111** [whutug/whu-thesis#147] [Q&A]
论文过学校抽检的时候学校要求给中英文摘要和目录的用罗马数字页码进行编号？我知道这与模板不符，qwq但请问有可能通过更改cls文件实现吗

**Comment by MZGuo111** [comment]
我仿照博士毕业论文在以下代码块添加了如下内容，可最后编译出来只有目录的最后一页有页码，请问这个怎么解决呢？
```
\cs_new_protected:Npn \__whu_make_front_matter_bachelor:
  {
    \__whu_make_cover_bachelor:
    \__whu_make_statesment_bachelor:

    \pagestyle { plain }
    \pagenumbering { Roman }
    \__whu_make_abstract:
    \__whu_make_contents:

    \pagestyle { plain }
    \pagenumbering { arabic }
    \__whu_set_baseline_skip:
    \raggedbottom
    \__whu_set_table_font:
  }
```

**Reply by xkwxdyy** [reply]
你是什么学位的论文，硕士？

**Reply by MZGuo111** [reply]
是本科的

**Reply by xkwxdyy** [reply]
要求说详细点，哪几页是要的，从谁开始计数，到哪里罗马结束

**Reply by MZGuo111** [reply]
从中文摘要开始，然后英文摘要，然后目录结束就ok了

**Reply by xkwxdyy** [reply]
[whu-thesis.cls.zip](https://github.com/whutug/whu-thesis/files/8768056/whu-thesis.cls.zip)
你把旧的 cls 替换成这个新的再试试，效果可能不完全是你想要的，你先看看吧

**Reply by MZGuo111** [reply]
感谢！

**Reply by MZGuo111** [reply]
hhhhh，是鱼佬，这个我已经解决了，目录把\tocloftpagestyle改成plain可以解决
```
%%% ---- 目录样式设置 ----- %%%
\RequirePackage { tocloft }
\clist_map_inline:nn { toc , lof , lot }
  {
    \tl_set:cn { cft #1 title font } { \hfil \heiti \zihao { -2 } }
    % 目录标题 黑体小 2 号
    \clist_map_inline:nn { before , after }
      { \skip_zero:c { cft ##1 #1 title skip } }
  }
\skip_set:Nn \cftaftertoctitleskip { 0.5 \baselineskip }

\clist_map_inline:nn { doctor , bachelor }
  {
    \tl_if_eq:NnT \g__whu_option_type_tl {#1}
      { \tocloftpagestyle { plain } }
  }
```

**Reply by Terrence-zzh** [reply]
十分感谢，首先看了三四遍没看懂，仅尝试了最新的回答没成功，阅读完whu-thesis.cls搞懂了。对本科论文来说，先让statement开始的页面样式设置为plain，页码风格为Roman；此时会发现摘要和目录都有一部分页没有对应的罗马下标，原因是cls中专门针对目录和摘要部分重新声明了页面样式指令为empty，所以会看不到。目录定位到上面的回答里，改为plain，此时摘要还是缺失罗马页码；
% 生成摘要
\cs_new_protected:Npn \__whu_make_abstract:
  {
    % 中文摘要
    \tl_if_empty:NF \g__whu_element_abstract_tl
      {
        \chapter * { 摘 \qquad 要 }
        \phantomsection
        \clist_map_inline:nn { doctor , master }
          {
            \tl_if_eq:NnT \g__whu_option_type_tl {##1}
              { \addcontentsline { toc } { chapter } { 摘 \qquad 要 } }
          }
        \tl_if_eq:NnT \g__whu_option_type_tl { bachelor }
          { \thispagestyle { plain } }
        \__whu_set_baseline_skip:

        \file_input:V \g__whu_element_abstract_tl

        \skip_vertical:N \c__whu_baseline_skip \par
        \__whu_put_keywords:nn
          { \heiti \zihao { -4 } 关键词：\hbox:n {} }
          { \clist_use:Nn \g__whu_info_keywords_clist { ； } }
      }
    % 英文摘要
    \tl_if_empty:NF \g__whu_element_abstract_en_tl
      {
        \chapter * { \textbf { ABSTRACT } }
        \phantomsection
        \clist_map_inline:nn { doctor , master }
          {
            \tl_if_eq:NnT \g__whu_option_type_tl {##1}
              { \addcontentsline { toc } { chapter } { ABSTRACT } }
          }
        \tl_if_eq:NnT \g__whu_option_type_tl { bachelor }
          { \thispagestyle { plain } }
        \__whu_set_baseline_skip:
        \file_input:V \g__whu_element_abstract_en_tl
        \skip_vertical:N \c__whu_baseline_skip \par
        \__whu_put_keywords:nn
          { \bfseries \zihao { -4 } Key ~ words: ~ }
          { \clist_use:Nn \g__whu_info_keywords_en_clist { ; ~ } }
      }
    \__whu_new_chapter_page:
  }
定位到这个部分，将两个的\thispagestyle由empty转换成plain，就可以了
